tithemi/assignment4/islands.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <array>
+#include <vector>
+#include <set>
+#include <iostream>
+
+template<size_t Rows, size_t Columns>
+size_t count_islands(const std::array<std::array<bool, Columns>, Rows>& map)

```

Comment:
As mentioned below, your algorithm is not entirely clear to me on the first glance.
Can you please add a couple of lines of comments describing the general, high-level approach to this problem?